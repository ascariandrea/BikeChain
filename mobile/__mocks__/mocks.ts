export const navigation = {
  setParams: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  replace: jest.fn(),
  popToTop: jest.fn(),
  isFocused: () => true,
  dangerouslyGetParent: jest.fn(),
  state: {
    index: 0,
    routeName: 'login',
    routes: [],
    isTransitioning: false,
    key: 'login',
    params: {}
  },
  navigate: jest.fn(),
  goBack: jest.fn(),
  dispatch: jest.fn(),
  closeDrawer: jest.fn(),
  openDrawer: jest.fn(),
  toggleDrawer: jest.fn(),
  getParam: jest.fn(),
  dismiss: jest.fn(),
  addListener: jest.fn()
};
