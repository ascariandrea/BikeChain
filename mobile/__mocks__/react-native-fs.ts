export default {
  mkdir: jest.fn(),
  moveFile: jest.fn(),
  copyFile: jest.fn()
};
