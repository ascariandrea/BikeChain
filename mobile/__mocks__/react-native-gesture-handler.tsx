import * as React from 'react';

export default {
  UIManager: {
    directEventTypes: jest.fn()
  }
};

class BaseButton extends React.Component<{}> {
  public render() {
    return React.createElement('BaseButton', this.props, this.props.children);
  }
}

export { BaseButton };
