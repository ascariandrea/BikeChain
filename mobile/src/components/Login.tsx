import { none, Option } from 'fp-ts/lib/Option';
import { Button, Form, Input, Item, Text } from 'native-base';
import * as React from 'react';
import { declareCommands } from 'react-avenger';
import { NavigationScreenProps } from 'react-navigation';
import { apiCommands } from '../commands';
import { ROUTES } from '../routes/routes';
import { fromEmptyString } from '../utils/utils';
import { FlexView } from './common';

const commands = declareCommands({
  login: apiCommands.login
});

type Props = typeof commands.Props & NavigationScreenProps;

interface State {
  email: Option<string>;
  password: Option<string>;
}
class Login extends React.Component<Props, State> {
  public state: State = {
    email: none,
    password: none
  };

  public onLoginPress = (email: string, password: string) => {
    this.props.login({ email, password }).then(result => {
      if (result.isRight()) {
        this.props.navigation.navigate(ROUTES.DEVICES);
      }
    });
  };
  public render() {
    const { email, password } = this.state;
    const emailValue = email.getOrElse('');
    const passwordValue = password.getOrElse('');

    return (
      <FlexView direction="row">
        <Form>
          <Item>
            <Input
              placeholder="Username"
              value={emailValue}
              onChangeText={t => {
                this.setState({
                  email: fromEmptyString(t)
                });
              }}
            />
          </Item>
          <Item last>
            <Input
              placeholder="Password"
              value={passwordValue}
              onChangeText={t => {
                this.setState({
                  password: fromEmptyString(t)
                });
              }}
              secureTextEntry={true}
            />
          </Item>
          <Button onPress={() => this.onLoginPress(emailValue, passwordValue)}>
            <Text>LOGIN</Text>
          </Button>
        </Form>
      </FlexView>
    );
  }
}

export default commands(Login);
