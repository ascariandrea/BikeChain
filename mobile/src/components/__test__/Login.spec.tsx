import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { navigation } from '../../../__mocks__/mocks';
import Login from '../Login';

test('should renders correctly', () => {
  const appComponent = renderer.create(<Login navigation={navigation} />);
  const tree = appComponent.toJSON();
  expect(tree).toMatchSnapshot();
});
