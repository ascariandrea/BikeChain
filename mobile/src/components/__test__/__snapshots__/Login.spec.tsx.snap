// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRadius": undefined,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginLeft": 15,
          "opacity": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="Username"
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "paddingLeft": 5,
            "paddingRight": 5,
            "top": 1.5,
          }
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value="me@bikechain.com"
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRadius": undefined,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginLeft": 0,
          "opacity": 1,
          "paddingLeft": 15,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "paddingLeft": 5,
            "paddingRight": 5,
            "top": 1.5,
          }
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value="password"
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        LOGIN
      </Text>
    </View>
  </View>
</View>
`;
